EKF/control.cpp

```diff
@@ -757,10 +757,19 @@ void Ekf::controlHeightFusion()
 		// we have just switched to using range finder, calculate height sensor offset such that current
 		// measurment matches our current height estimate
 		if (_control_status_prev.flags.rng_hgt != _control_status.flags.rng_hgt) {
-			if (_terrain_initialised) {
-				_hgt_sensor_offset = _terrain_vpos;
+			// use the parameter rng_gnd_clearance if on ground to avoid a noisy offset initialization (e.g. sonar)
+			if (_control_status.flags.in_air) {
+
+				if (_terrain_initialised) {
+
+					_hgt_sensor_offset = _terrain_vpos;
+				} else {
+
+					_hgt_sensor_offset = _R_rng_to_earth_2_2 * _range_sample_delayed.rng + _state.pos(2);
```

Comment:
A per https://dev.px4.io/en/advanced/parameter_reference.html#EKF2_HGT_MODE, I do not think we should update the hgt offset when range sensor is the primary sensor (though it makes perfect sense when baro or gps is primary).
Shouldn't it read like this?
```cpp
		// we have just switched to using range finder, calculate height sensor offset such that current
		// measurment matches our current height estimate
		if (_control_status_prev.flags.rng_hgt != _control_status.flags.rng_hgt) {
			// use the parameter rng_gnd_clearance if on ground to avoid a noisy offset initialization (e.g. sonar)
			if (_control_status.flags.in_air && _terrain_initialised) {
				hgt_sensor_offset = _terrain_vpos;

			} else {

				_hgt_sensor_offset = _params.rng_gnd_clearance;
			}
		}
```